src/components/indicatorMenu/ColorScaleIcon.vue

```diff
@@ -23,8 +24,18 @@ const props = defineProps({
 });
 
 const halfway = computed(() => props.size / 2);
-const { colorScales } = useColorScale();
+const { getIndicatorValueColor, getIndicatorColorType, getIndicatorColorCategories } = useIndicatorColors();
 
-const colorScale = computed(() => colorScales.value[props.indicator]);
-const getColor = (value) => colorScale.value(value);
+const iconColors = computed(() => {
+    if (getIndicatorColorType(props.indicator) === ColorType.Scale) {
+        const colorScaleSampleValues = [0.7, 0.9, 0.4, 0.1];
+        return colorScaleSampleValues.map((value) => getIndicatorValueColor(props.indicator, value, false));
+    }
+    const result = [];
+    const categoryColors = getIndicatorColorCategories(props.indicator).map((cat) => cat.color);
+    while (result.length < 4) {
+        result.push(...categoryColors);
+    }
```

Comment:
I'll put a check in and throw an error if length is 0 - we're already making some assumptions that the colour categories are configured sensibly, but you're right, this would be a particularly bad one to hit!